otx/algorithms/action/adapters/openvino/model_wrappers/openvino_models.py

```diff
@@ -0,0 +1,185 @@
+    """Get multiclass predictions."""
+    index = np.argmax(logits)
+    if activate:
+        logits = softmax_numpy(logits)
+    return [(index, logits[index])]
+
+
+# pylint: disable=too-many-instance-attributes
+class OTXActionCls(Model):
+    """OTX Action Classification model for openvino task."""
+
+    __model__ = "ACTION_CLASSIFICATION"
+
+    def __init__(self, model_adapter, configuration=None, preload=False):
+        """Image model constructor.
+
+        Calls the `Model` constructor first
+        """
+        super().__init__(model_adapter, configuration, preload)
+        self.image_blob_names, self.image_info_blob_names = self._get_inputs()
+        self.image_blob_name = self.image_blob_names[0]
+        self.out_layer_name = self._get_outputs()
+
+        _, self.n, self.c, self.t, self.h, self.w = self.inputs[self.image_blob_name].shape
+        self.resize = RESIZE_TYPES["standard"]
+        self.input_transform = InputTransform(False, None, None)
+
+        # FIXME Below parameters should be changed dynamically from data pipeline config
+        self.interval = 4
+
+    def _get_inputs(self):
+        image_blob_names, image_info_blob_names = [], []
+        for name, metadata in self.inputs.items():
+            if len(metadata.shape) == 6:
+                image_blob_names.append(name)
+            elif len(metadata.shape) == 4:
+                image_info_blob_names.append(name)
+        return image_blob_names, image_info_blob_names
+
+    def _get_outputs(self):
+        layer_name = "logits"
+        for name, meta in self.outputs.items():
+            if "logits" in meta.names:
+                layer_name = name
+        return layer_name
+
+    @check_input_parameters_type()
+    def preprocess(self, inputs: List[DatasetItemEntity]):
+        """Pre-process."""
+        meta = {"original_shape": inputs[0].media.numpy.shape}
+        frames = []
+        for item in inputs:
+            frame = item.media.numpy
+            frame = self.resize(frame, (self.w, self.h))
+            frames.append(frame)
+        np_frames = self.reshape(frames)
+        dict_inputs = {self.image_blob_name: np_frames}
+        meta.update({"resized_shape": np_frames[0].shape})
+        return dict_inputs, meta
+
+    @staticmethod
+    @check_input_parameters_type()
+    def reshape(inputs: List[np.ndarray]) -> np.ndarray:
+        """Reshape(expand, transpose, permute) the input np.ndarray."""
+        np_inputs = np.expand_dims(inputs, axis=(0, 1))  # [1, 1, T, H, W, C]
+        np_inputs = np_inputs.transpose(0, 1, -1, 2, 3, 4)  # [1, 1, C, T, H, W]
+        return np_inputs
+
+    @check_input_parameters_type()
+    # pylint: disable=unused-argument
+    def postprocess(self, outputs: Dict[str, np.ndarray], meta: Dict[str, Any]):
+        """Post-process."""
+        logits = outputs[self.out_layer_name].squeeze()
+        return get_multiclass_predictions(logits)
+
+
+class OTXActionDet(OTXActionCls):
```

Comment:
Is it a proper inheritance? I think `Cls` and `Det` have an independent relationship.

Reply:
It is for reducing duplicated code. Do you think it's better to separate both models?

Reply:
If so, how about making base class to reduce the duplication?

Reply:
It's better to separate them. Please check once again